### ISSUE TITLE
Move documentation to GH pages?

### ISSUE THREAD

**Issue by steven-murray** [RadioAstronomySoftwareGroup/pyuvdata#768]
@bhazelton and I were discussing whether moving the documentation building to GH pages might be useful. Some advantages are the clear build process, and the fact that everything stays in the repo, as well as being able to build on PRs automatically (and check the build). Also the build can be controlled through GH actions.

An example action: https://github.com/arokem/pyAFQ/blob/master/.github/workflows/docbuild.yml

**Comment by bhazelton** [comment]
Another problem that just came up is that the RTD build is currently failing because it's using too much memory to create the conda environment.

**Comment by bhazelton** [comment]
I now have an implementation of this working on pyradiosky, so I know how to do it.

**Comment by bhazelton** [comment]
I think we should not do this. I've experimented with this on pyradiosky and I got something working but it's not as good as RTD for several reasons:
- No easy way to have docs for different versions (might be able to do this with folders but then we need a top level landing page with links to the different versions and then the top level landing page won't have the readme content).
- No good way to host docs temporarily for a branch that's going through review and then remove them when the branch is merged into master. This is possible on RTD (although it requires going to the site and turning on the branch in question).
- The push to gh-pages seems to be a bit flakey. I'm not sure why, but it periodically errors and then fixes itself.

I could see still using github actions to build the docs just as a check that the build passes, but I don't think hosting them on GitHub pages offers as much as RTD does (at least currently).
